### PR TITLE
[#173218638] Handle promise rejection in mailup transport

### PR DESCRIPTION
**Pull request by cloudify**

### CODE REVIEW

File: src/utils/mailup.ts [hunk 1]
```diff
@@ -293,21 +293,20 @@ export function MailUpTransport(
...
-      if (isRight(errorOrResponse)) {
-        // tslint:disable-next-line:no-null-keyword
-        return callback(null, {
-          ...errorOrResponse.value,
-          messageId: mail.data.messageId
-        });
-      } else {
-        return callback(errorOrResponse.value, undefined);
-      }
+      sendTransactionalMail(options.creds, email, fetchAgent)
```
Comment (gunzip) [inline src/utils/mailup.ts:296]: just note that `sendTransactionalMail` cannot throw any exception

Comment (cloudify) [reply src/utils/mailup.ts:296]: you never know :)

Comment (gunzip) [inline src/utils/mailup.ts:296]: do we need a `return` here ?

Comment (cloudify) [reply src/utils/mailup.ts:296]: nope, [the caller](https://github.com/nodemailer/nodemailer/blob/7750e048ff79d099500061497fb45b2861a7652c/lib/mailer/index.js#L214) doesn't expect a return value

**Pull request merged**